### PR TITLE
fix: Localized String in NewLoginAlert we show in SelfProfile 

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

In #844 I did a mistake and passed the wrong parameter in the conctruction of a localized string which led to show the wrong text inside an alert. While I figured that out I did some more testing and realized that even before that change we were showing the text but in the wrong format. 

Screenshot 2024-02-01 at 12 15 43

With this PR I fix both of those mistakes and add some unit tests. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by caldrian — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/SelfProfileViewControllerTests.swift
Comment (agisilaos) [inline wire-ios/Wire-iOS Tests/SelfProfileViewControllerTests.swift:1]: I know that in this file we have both Unit and SnapshotTests. I would like to separate those in a different PR.

File: wire-ios-data-model/Source/Model/UserClient/UserClientType.swift [hunk 1]
```diff
@@ -37,6 +37,8 @@ import Foundation
     /// True if the self user has verfied the client
     var verified: Bool { get }
 
+    var needsToNotifyUser: Bool { get set }
+
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/Model/UserClient/UserClientType.swift:41]: question: I don't see the variable `needsToNotifyUser` used in the changes, why was it added?

Comment (agisilaos) [reply wire-ios-data-model/Source/Model/UserClient/UserClientType.swift:41]: Good question! When I changed the type from UserClient to UserClientType I would get an error in this method:

```swift
fileprivate func presentNewLoginAlertController(_ clients: [UserClientType]) {
        let newLoginAlertController = UIAlertController(forNewSelfClients: clients)

        let actionManageDevices = UIAlertAction(title: L10n.Localizable.Self.NewDeviceAlert.manageDevices, style: .default) { _ in
            self.openControllerForCellWithIdentifier(SettingsCellDescriptorFactory.settingsDevicesCellIdentifier)
        }

        newLoginAlertController.addAction(actionManageDevices)

        let actionTrustDevices = UIAlertAction(title: L10n.Localizable.Self.NewDeviceAlert.trustDevices, style: .default) { [weak self] _ in
            self?.presentUserSettingChangeControllerIfNeeded()
        }

        newLoginAlertController.addAction(actionTrustDevices)

        present(newLoginAlertController, animated: true, completion: .none)

        userSession.enqueue {
            clients.forEach {
                $0.needsToNotifyUser = false
            }
        }
    }
```

More specifically in line:

**$0.needsToNotifyUser = false**

needsToNotifyUser wasn't part of UserClientType and I needed to add it.

Comment (markusfassbender) [reply wire-ios-data-model/Source/Model/UserClient/UserClientType.swift:41]: thanks!

**Review by markusfassbender — APPROVED** [review]
